src/contrast.rs

```diff
@@ -364,6 +364,7 @@ mod tests {
         }
     }
 
+    #[cfg_attr(miri, ignore)]
```

Comment:
Opting out for each benchmark individually feels a bit error prone - could you do the tests vs benches module split everywhere?

Reply:
Sure, just didn't want to scare you with code moves all over the place

Reply:
Then let's do it in the second PR.